conans/client/tools/scm.py

```diff
@@ -89,23 +99,34 @@ def run(self, command):
         command = self._configure_ssl_verify + command
         return super(Git, self).run(command)
 
-    def clone(self, url, branch=None, args=""):
+    def clone(self, url, element=None, args="", shallow=False):
         url = self.get_url_with_credentials(url)
         if os.path.exists(url):
             url = url.replace("\\", "/")  # Windows local directory
-        if os.path.exists(self.folder) and os.listdir(self.folder):
-            if not branch:
+        if (os.path.exists(self.folder) and os.listdir(self.folder)) or (shallow and element):
+            if not element:
                 raise ConanException("The destination folder '%s' is not empty, "
                                      "specify a branch to checkout (not a tag or commit) "
                                      "or specify a 'subfolder' "
                                      "attribute in the 'scm'" % self.folder)
+            mkdir(self.folder)
```

Comment:
Why the mkdir here and now?

Reply:
I think just leftover, I'll remove, should work without it

Reply:
removed